File: web/src/types/post.ts
// types/post.ts

/** 创建根帖请求参数 */
export interface CreateRootPostPayload {
  nodeSlug: string;
  title: string;
  content: string;
  tags?: string[];
  imageList?: string[];
}

export interface UpdateRootPostPayload {
  nodeSlug: string;
  title: string;
  content: string;
  tags?: string[];
}

/** 创建回复请求参数（不再包含 parentSlug，由 URL 路径承载） */
export interface CreateReplyPayload {
  content: string;
  imageList?: string[];
}
export interface UpdateReplyPayload {
  content: string;
}

export interface CreatePostResponse {
  slug: string;
}

File: api/form/post.go
package form

// RootPostCreateForm 创建根帖专用
type RootPostCreateForm struct {
	NodeSlug  string   `json:"nodeSlug" binding:"required"`
	Title     string   `json:"title" binding:"required,min=5,max=100"`
	Content   string   `json:"content" binding:"required"`
	Tags      []string `json:"tags"`
	ImageList string `json:"imageList"`

	// 以下字段由 Controller 内部注入，不参与 JSON 绑定与校验
	UserSlug  string   `json:"-"`
}

// RootPostUpdateForm root post update form
type RootPostUpdateForm struct {
	Slug 	 string   `uri:"slug" json:"slug" binding:"required"`
	Title    string   `form:"title" json:"title"`
	Content  string   `form:"content" json:"content"`
	NodeSlug string   `form:"nodeSlug" json:"nodeSlug"`
	Tags     []string `form:"tags" json:"tags"`
}

// ReplyCreateForm 回复专用
type ReplyCreateForm struct {
	Slug 	   string   `uri:"slug" json:"slug" binding:"required"`
	Content    string   `json:"content" binding:"required"`
	ImageList  string   `json:"imageList"`

	// 以下字段由 Controller 内部注入，不参与 JSON 绑定与校验
	Title      string   `json:"-"`
	ParentSlug string   `json:"-"`
	UserSlug   string   `json:"-"`
}

// ReplyUpdateForm reply update form
type ReplyUpdateForm struct {
	Slug 	 string   `uri:"slug" json:"slug" binding:"required"`
	Content  string   `form:"content" json:"content"`

	Title    string   `json:"-"`
}
